src/openrct2/paint/sprite/Paint.Misc.cpp

```diff
@@ -38,6 +38,8 @@ void misc_paint(paint_session* session, const SpriteBase* misc, int32_t imageDir
         case SPRITE_MISC_STEAM_PARTICLE: // 0
         {
             auto particle = misc->As<SteamParticle>();
+            if (particle == nullptr)
+                return;
```

Comment:
Are all of this really needed? Because if the type already matches, `Is<>` will return true and it will have something that isn't a `nullptr` anyway

Reply:
Compilers unfortunately weren't intelligent enough to know that it wouldn't be a nullptr.

Reply:
I thought so :\